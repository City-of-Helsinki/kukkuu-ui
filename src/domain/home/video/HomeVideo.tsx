import { useTranslation } from 'react-i18next';

import styles from './homeVideo.module.scss';
import { getCurrentLanguage } from '../../../common/translation/TranslationUtils';

const HomeVideo = () => {
  const { t, i18n } = useTranslation();
  const currentLanguage = getCurrentLanguage(i18n);

  const videoSrc = `https://www.youtube.com/embed/${t(
    'home.video.youtubeId'
  )}?controls=0&fs=0&hl=${currentLanguage}&autoplay=1`;

  return (
    <section className={styles.wrapper}>
      <div className={styles.innerwrapper}>
        <div className={styles.embedContainer}>
          <iframe
            id="ytplayer"
            title="video"
            src={videoSrc}
            allowFullScreen
            frameBorder="0"
          ></iframe>
        </div>
      </div>
    </section>
  );
};

export default HomeVideo;
