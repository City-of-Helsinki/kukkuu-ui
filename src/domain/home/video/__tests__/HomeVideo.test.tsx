import { render } from '@testing-library/react';

import HomeVideo from '../HomeVideo';

it('renders snapshot correctly', () => {
  const { container } = render(<HomeVideo />);
  expect(container).toMatchSnapshot();
});
