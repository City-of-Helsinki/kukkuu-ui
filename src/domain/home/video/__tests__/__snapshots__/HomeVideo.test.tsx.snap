// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<section
  className="wrapper"
>
  <div
    className="innerwrapper"
  >
    <div
      className="embedContainer"
    >
      <iframe
        allowFullScreen={true}
        frameBorder="0"
        id="ytplayer"
        src="https://www.youtube.com/embed/jYteckzo1uA?controls=0&fs=0&hl=fi&autoplay=1"
        title="video"
      />
    </div>
  </div>
</section>
`;
